subworkflows/local/utils_nfcore_scrnaseq_pipeline/main.nf

```diff
@@ -80,25 +80,48 @@ workflow PIPELINE_INITIALISATION {
     //
     // Create channel from input file provided through params.input
     //
-    Channel
-        .fromSamplesheet("input")
-        .map {
-            meta, fastq_1, fastq_2 ->
-                if (!fastq_2) {
-                    return [ meta.id, meta + [ single_end:true ], [ fastq_1 ] ]
-                } else {
-                    return [ meta.id, meta + [ single_end:false ], [ fastq_1, fastq_2 ] ]
-                }
-        }
-        .groupTuple()
-        .map {
-            validateInputSamplesheet(it)
-        }
-        .map {
-            meta, fastqs ->
-                return [ meta, fastqs.flatten() ]
-        }
-        .set { ch_samplesheet }
+    if (params.aligner == 'cellrangermulti') { // the cellrangermulti sub-workflow logic needs that channels have reads separated by feature_type. Cannot merge all.
+        Channel
+            .fromSamplesheet("input")
+            .map {
+                meta, fastq_1, fastq_2 ->
+                    if (!fastq_2) {
+                        return [ meta.id, meta.feature_type, meta + [ single_end:true ], [ fastq_1 ] ]
+                    } else {
+                        return [ meta.id, meta.feature_type, meta + [ single_end:false ], [ fastq_1, fastq_2 ] ]
+                    }
+            }
+            .groupTuple( by: [0,1] )
+            .map{ id, type, meta, reads -> [ id, meta, reads ] }
+            .map {
+                validateInputSamplesheet(it)
+            }
+            .map {
+                meta, fastqs ->
+                    return [ meta, fastqs.flatten() ]
+            }
+            .set { ch_samplesheet }
+    } else {
+        Channel
+            .fromSamplesheet("input")
+            .map {
+                meta, fastq_1, fastq_2 ->
+                    if (!fastq_2) {
+                        return [ meta.id, meta + [ single_end:true ], [ fastq_1 ] ]
+                    } else {
+                        return [ meta.id, meta + [ single_end:false ], [ fastq_1, fastq_2 ] ]
+                    }
+            }
+            .groupTuple()
+            .map {
+                validateInputSamplesheet(it)
+            }
+            .map {
+                meta, fastqs ->
+                    return [ meta, fastqs.flatten() ]
+            }
+            .set { ch_samplesheet }
+    }
```

Comment:
Hi @grst ,
I do not know if I follow. You mean not having the "else" part for the non-cellranger multi ones?

Cellranger multi itself needs it, because the module, expects the inputs as separate channels, one for each data type, in order to properly generate the execution configuration file.

The different-input-channel handling goes [here](https://github.com/nf-core/scrnaseq/blob/247-support-for-10x-ffpe-scrna/subworkflows/local/align_cellrangermulti.nf#L64-L105) so the module itself can receive what it expects.

Reply:
> You mean not having the "else" part for the non-cellranger multi ones?

Yes, i.e. would it harm for the other aligners to also have the sample grouped by feature type? If all samplese are "gex" then this additional grouping shouldn't make any difference, no?

Reply:
Ah, I see, and agree.
But, I rather think this might be a new PR. As it may also involve nf-tests updates, and all.

Reply:
I will create a ticket for that.
I will link it in my next summary comment where I will add the changes included for handling `FFPE` testing dataset, including the nf-core module update.
Then I resolve this thread.

Reply:
Hi @grst ,
Using this ticket #272 for tracking these.
Already added there this one.

Resolving this thread.